src/domotica/domotica.c

```diff
@@ -22,48 +22,52 @@ __attribute__ ((weak, alias ("domotica_handle_output_change_dummy"))) \
 
 // ------------------------------------------------------------------
 // Internal structure of a queue to maintain all changes that should be handled
-typedef struct OUTPUT_CHANGE {
+typedef struct {
   uint16_t mask_on;
   uint16_t mask_off;
-  struct OUTPUT_CHANGE *next;
 } OUTPUT_CHANGE_Type;
 
-static OUTPUT_CHANGE_Type *output_change_queue = 0;
+typedef struct {
+  OUTPUT_CHANGE_Type buffer[DOMOTICA_CHANGE_BUFFER_Size];
+  volatile uint8_t writer;
+  volatile uint8_t reader;
+} OUTPUT_CHANGE_RINGBUFFER_Type;
```

Comment:
I changed the names, as it only appears in the top part. Also the OUTPUT_CHANGE_Type needs to be exported for the next release to add priorities to it...